src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.PropertyDescriptorGridEntryAccessibleObjectTests.cs

```diff
@@ -0,0 +1,195 @@
+            gridEntry.parentPE = new TestGridEntry(propertyGrid, null, testPropertyGridView);
+
+            var firstChild = gridEntry.AccessibilityObject.FragmentNavigate(Interop.UiaCore.NavigateDirection.FirstChild);
+            Assert.NotNull(firstChild);
+            Assert.Equal(typeof(PropertyGridView.DropDownHolder.DropDownHolderAccessibleObject), firstChild.GetType());
+        }
+
+        [WinFormsFact]
+        public void PropertyDescriptorGridEntryAccessibleObject_Ctor_Default()
+        {
+            using var propertyGrid = new PropertyGrid();
+            var propertyDescriptorGridEntryTestEntity = new TestPropertyDescriptorGridEntry(propertyGrid, null, false);
+            var propertyDescriptorGridEntryAccessibleObject = propertyDescriptorGridEntryTestEntity.AccessibilityObject;
+
+            Assert.NotNull(propertyDescriptorGridEntryAccessibleObject);
+
+            TypeInfo propertyDescriptorGridEntryAccessibleObjectTypeInfo = propertyDescriptorGridEntryAccessibleObject.GetType().GetTypeInfo();
+            FieldInfo owningPropertyDescriptorGridEntryField = propertyDescriptorGridEntryAccessibleObjectTypeInfo.GetDeclaredField("_owningPropertyDescriptorGridEntry");
+            var owningGridEntry = owningPropertyDescriptorGridEntryField.GetValue(propertyDescriptorGridEntryAccessibleObject);
+
+            Assert.Equal(propertyDescriptorGridEntryTestEntity, owningGridEntry);
+        }
+
+        [WinFormsFact]
+        public void PropertyDescriptorGridEntryAccessibleObject_ExpandCollapseState_collapsed_by_default()
+        {
+            using var propertyGrid = new PropertyGrid();
+            var propertyDescriptorGridEntryTestEntity = new TestPropertyDescriptorGridEntry(propertyGrid, null, false);
+            var propertyDescriptorGridEntryAccessibleObject = propertyDescriptorGridEntryTestEntity.AccessibilityObject;
+
+            var expandCollapseState = propertyDescriptorGridEntryAccessibleObject.ExpandCollapseState;
+            Assert.Equal(UiaCore.ExpandCollapseState.Collapsed, expandCollapseState);
+        }
+
+        [WinFormsFact]
+        public void PropertyDescriptorGridEntryAccessibleObject_ExpandCollapseState_reflects_ExpandablePropertyState()
+        {
+            using Form form = new Form();
+            using PropertyGrid propertyGrid = new PropertyGrid();
+            var testEntity = new TestEntity();
+            testEntity.FontProperty = new Font(FontFamily.GenericSansSerif, 1);
+            propertyGrid.SelectedObject = testEntity;
+
+            form.Controls.Add(propertyGrid);
+
+            Type propertyGridType = typeof(PropertyGrid);
+            FieldInfo[] fields = propertyGridType.GetFields(
+                BindingFlags.NonPublic |
+                BindingFlags.Instance);
+            FieldInfo gridViewField = fields.First(f => f.Name == "_gridView");
+            PropertyGridView propertyGridView = gridViewField.GetValue(propertyGrid) as PropertyGridView;
```

Comment:
- We already have TestAccessor for these cases. So please use it everywhere in your tests for non-pablic members.
- Use type cast instead of using **as** operator to avoid getting NRE in the next. Use it for such places everywhere in your tests.
```suggestion
            PropertyGridView propertyGridView = (PropertyGridView)propertyGrid.TestAccessor().Dynamic._gridView;
```